script.js

```diff
@@ -1,4 +1,181 @@
+    if (isNaN(numberOfDice) || input == "") {
+      return `Not a number! Please select number of dices to roll`;
+    }
+    //Move to next game mode
+    gameState = "game";
+    return `All players will now be playing with ${numberOfDice} dices`;
+  }
+
+  //Game Function
+  if (gameState == "game") {
+    //Option for user to change game mode halfway
+    if (input == "change") {
+      gameMode = "";
+      return `Change Mode! Please choose highest or lowest`;
+    }
+
+    while (playerNumber <= numberOfPlayers) {
+      var normal = normalGame();
+      return normal;
+    }
+    //Move to next game state
+    gameState = "checkWinner";
+  }
+
+  //Game state to check for winner
+  if (gameState == "checkWinner") {
+    //Check for winner
+    var winner = checkWinner(numberOfPlayers, playerScore);
+    //Print leaderboard
+    var message = `Leaderboard<br><br>Current Game Mode: ${gameMode}<br><br><b>Current Winner:</b> Player ${winner[0][1]} with a score of ${winner[0][0]}!!🎉🎉<br><br>PlayerNo.: PlayerScore<br>`;
+
+    for (i = 0; i < numberOfPlayers; i++) {
+      message += `<br>Player ${winner[i][1]}: ${winner[i][0]}`;
+    }
+    //Reset the game
+    gameState = "game";
+    playerNumber = 1;
+
+    return message;
+  }
+  return output;
 };
+
+//Check Winner & Prints leaderboard in decreasing order
+function checkWinner(player, input1) {
+  for (i = 0; i < player; i++) {
+    leaderBoard[i] = [];
+  }
+  //Create array for player number and corresponding player score
+  for (j = 0; j < player; j++) {
+    leaderBoard[j] = [input1[j], [j + 1]];
+  }
+
+  //Sort array by 1st element, player score
+  if (gameMode == "highest") {
+    leaderBoard.sort().reverse();
+  } else {
+    leaderBoard.sort();
+  }
+  return leaderBoard;
+}
+
+// Check Order automatically by game mode
+function checkOrder(playerList) {
+  var list = [];
+  var length = playerList.length;
+  if (gameMode == "highest") {
+    //Sort dice rolls in descending order
+    playerList.sort();
+
+    for (var counter = 0; counter < length; counter++) {
+      list += playerList[counter];
+    }
+    list = parseInt(list.toString());
+    return list;
+  }
+  if (gameMode == "lowest") {
+    //Sort dice rolls in ascending order
+    playerList = playerList.sort().reverse();
+    for (var counter = 0; counter < length; counter++) {
+      list += playerList[counter];
+    }
+    list = parseInt(list.toString());
+    return list;
+  }
+}
+
+//Dice roll function
+function diceRoll() {
+  return Math.floor(Math.random() * 6) + 1;
+}
+
+//Normal game function
+function normalGame() {
+  var playersSum = [];
+  //For every player, roll dice, check order of dice rolls and score
+  while (playerNumber <= numberOfPlayers) {
+    //Roll dice as per numberOfDices
+    var diceList = [];
+    for (counter = 0; counter < numberOfDice; counter++) {
+      diceList.push(diceRoll());
+    }
+
+    //Check for best number order
+    playersSum[playerNumber - 1] = checkOrder(diceList);
```

Comment:
I like how you are utilising the `checkOrder` function, and automating the best choice for player